pkg/controller/kieapp/defaults/defaults.go

```diff
@@ -1285,6 +1285,28 @@ func setPasswords(spec *api.KieAppSpec, isTrialEnv bool) {
 	}
 }
 
+func handleDatagridAuth(spec *api.KieAppSpec, status api.KieAppStatus, passwords []*string) []*string {
+	if spec.Objects.Console != nil && spec.Objects.Console.DataGridAuth != nil && len(spec.Objects.Console.DataGridAuth.Password) == 0 {
+		if status.Applied.Objects.Console != nil && status.Applied.Objects.Console.DataGridAuth != nil && len(status.Applied.Objects.Console.DataGridAuth.Password) > 0 {
+			spec.Objects.Console.DataGridAuth.Password = status.Applied.Objects.Console.DataGridAuth.Password
```

Comment:
shouldn't be also here set added password to passwords? Or it is already handled elsewhere?

Reply:
Here we are appending to the array of passwords if needed, depends on the information in the spec/status then are checked here https://github.com/kiegroup/kie-cloud-operator/pull/779/files/86e3d039a521ac7762be044765803d1e1453c21a#diff-29ad51b906bc49dd54db094a3b626e8e4a04628d03562e6c0d45db0e391ac2fbR1276

Reply:
Nope, we add to the array of passwords if a password must be included in the deploy and is empty. On Datagrid we check if the object are nil or not, instead on the kieadmin (mandatory for every deploy) we add to generate only if is empty, otherwise we set what is already deployed.